pkg/collector/corechecks/snmp/internal/checkconfig/buildprofile.go

```diff
@@ -71,6 +71,7 @@ func (c *CheckConfig) BuildProfile(sysObjectID string) (profiledefinition.Profil
 		profile.Device.Vendor = rootProfile.Device.Vendor
 	}
 	profile.Metadata = updateMetadataDefinitionWithDefaults(profile.Metadata, c.CollectTopology, c.CollectVPN)
+	profile.Metrics = updateMetricsDefinitionWithDefaults(profile.Metrics, c.CollectVPN)
```

Comment:
Both for Metadata and Metrics, this means we'll try to fetch the VPN data on all devices even when it might not be available?
Could we figure out a way to _only_ fetch all these metrics and metadata if we know they are available?

Reply:
Comment resolved: this is solved in another PR by storing which OIDs don't have data